clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2949,12 +2945,15 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
         GetSYCLKernelObjectType(KernelCallerFunc)->getAsCXXRecordDecl();
 
     KernelCallOperatorVisitor KernelCallOperator(KernelCallerFunc, KernelObj);
-    KernelCallOperator.TraverseDecl(KernelCallerFunc);
+
     CXXMethodDecl *WGLambdaFn = nullptr;
-    if (KernelObj->isLambda())
+    if (KernelObj->isLambda()) {
       WGLambdaFn = KernelObj->getLambdaCallOperator();
-    else
-      WGLambdaFn = KernelCallOperator.CallOperator;
+    } else {
+      KernelCallOperator.TraverseDecl(KernelCallerFunc);
+      WGLambdaFn = KernelCallOperator.GetCallOperator();
+    }
```

Comment:
I think this whole if and else clause can be replaced with:
  WGLamdaFn = KernelCalloperator.getCallOperator();